packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -519,6 +520,18 @@ abstract class ImageStreamCompleter with Diagnosticable {
     _checkDisposed();
     _hadAtLeastOneListener = true;
     _listeners.add(listener);
+    _synchronouslyCallAddedListener(listener);
+  }
+
+  /// Add a listener for errors. It is similar to [addListener], but see
+  /// [removePeekListener] for a comparison of the behaviors.
+  void addPeekListener(ImageStreamListener listener) {
+    _checkDisposed();
+    _peekListeners.add(listener);
+    _synchronouslyCallAddedListener(listener);
+  }
+
+  void _synchronouslyCallAddedListener(ImageStreamListener listener) {
```

Comment:
nit: rename this method to something to imply that the listener is _only_ called when certain conditions are fulfilled, not unconditionally.